src/org/openlcb/cdi/jdom/CdiMemConfigReader.java

```diff
@@ -44,27 +44,37 @@ public CdiMemConfigReader(NodeID node, OlcbInterface iface, int space) {
 
     long nextAddress = 0;
     StringBuffer buf;
-    
+
     ReaderAccess retval;
     public void startLoadReader(ReaderAccess retval) {
         this.retval = retval;
         nextAddress = 0;
         buf = new StringBuffer();
         nextRequest();
     }
-    
+
     void nextRequest() {
         if (retval != null) {
             retval.progressNotify(buf.length(), -1);
         }
         MemoryConfigurationService.McsReadHandler memo =
             new MemoryConfigurationService.McsReadHandler() {
+
+                int retryCount = 0;
+
                 @Override
                 public void handleFailure(int code) {
-                    logger.warning("Error reading CDI: " + Integer.toHexString(code));
-                    done();
-                    // TODO: 5/2/16 proxy error messages to the caller.
-                    // don't do next request
+                     // see if we can retry
```

Comment:
We should check for the error code "read out of bounds". 
(I believe that's 0x1082: https://github.com/openlcb/documents/blob/master/standards/generated/MemoryConfigurationS.txt#L76 and https://github.com/openlcb/documents/blob/master/standards/generated/MessageNetworkS.txt#L375)

If the error that came back is out of bounds, then we don't need to re-try nor emit a warning, we are just done with the reads.
I guess we already do this in the memory tool when downloading a full memory space.

The way to test this is to have a memory space that's a multiple of 64 bytes long.

Reply:
Done, thanks for pointing that out.